pkg/apis/validation/validation.go

```diff
@@ -113,11 +115,59 @@ func boolValue(path *field.Path, s string) field.ErrorList {
 	return nil
 }
 
-func keyEncodingValue(path *field.Path, s string) field.ErrorList {
-	if s != string(cmapi.PKCS1) && s != string(cmapi.PKCS8) {
-		return field.ErrorList{field.NotSupported(path, s, []string{string(cmapi.PKCS1), string(cmapi.PKCS8)})}
+func keyValue(path *field.Path, attr map[string]string) field.ErrorList {
+	algPath := path.Child(csiapi.KeyAlgorithmKey)
+	encodingPath := path.Child(csiapi.KeyEncodingKey)
+	sizePath := path.Child(csiapi.KeySizeKey)
+
+	alg := attr[csiapi.KeyAlgorithmKey]
+	encoding := attr[csiapi.KeyEncodingKey]
+	size := attr[csiapi.KeySizeKey]
+
+	if encoding != string(cmapi.PKCS1) && encoding != string(cmapi.PKCS8) {
+		return field.ErrorList{field.NotSupported(encodingPath, encoding, []string{string(cmapi.PKCS1), string(cmapi.PKCS8)})}
+	}
+
+	switch alg {
```

Comment:
suggestion: Sort of related to my other comment - I think checking for exactly the `cmapi.*KeyAlgorithm` string is a bit strict and the UX isn't great. If I write `csi.cert-manager.io/key-algorithm: ecdsa` I kinda want that to work - it's not ambiguous, it's just the wrong casing.

Can we using `strings.ToUpper` on `alg` when making these comparisons, so that users don't have to remember the correct case to use when writing YAML?

Reply:
Instead of converting the case or doing case-insensitive comparisons everywhere, I normalize the user-provided values when we set the defaults so we can use the constants everywhere in the code-base.